java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -346,6 +357,28 @@ private boolean hasConstraintForDomain(LearnedConstraint lc, Class<? extends Con
       return flowBuilder.build();
     }
 
+    private JavaFileScannerContext.Location methodInvocationReturnMessage(Constraint constraint, ExplodedGraph.Edge edge, String methodName) {
+      String msg;
+      if (isConstraintOnlyPossibleResult(constraint, edge)) {
```

Comment:
If I didn't missed something, this seems to be tested only with `NULL` constraint, it would be good to add tests using other domains see example below

```
class ZeroConstraint {

  private int maybeZero() {
    if (b) {
      return 0;
    }
    return 1;
  }

  private int zero() {
    return 0;
  }

  void t1() {
    int i = maybeZero(); // flow@zero1 {{'maybeZero()' can return zero.}} flow@zero1 {{'i' is assigned zero.}}
    1 / i; // Noncompliant [[flows=zero1]] flow@zero1
  }

  void t() {
    int i = zero(); // flow@zero2 {{'zero()' returns zero.}} flow@zero2 {{'i' is assigned zero.}}
    1 / i; // Noncompliant [[flows=zero2]] flow@zero2
  }
}
```